vital/io/camera_from_metadata.h

```diff
@@ -0,0 +1,66 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief Functions to generate \ref kwiver::vital::camera_rpc from metadata
+ */
+
+#ifndef VITAL_CAMERA_FROM_METADATA_H_
+#define VITAL_CAMERA_FROM_METADATA_H_
+
+#include <vital/vital_export.h>
+
+#include <vital/types/camera_rpc.h>
+#include <vital/types/metadata.h>
+
+namespace kwiver {
+namespace vital {
+
+/// Convert space separated sting to Eigen vector
+/**
+ * \param s The string to be converted.
+ * \return The converted vector.
+ */
+Eigen::VectorXd
+VITAL_EXPORT string_to_vector( std::string const& s );
+
+/// Produce RPC camera from metadata
+/**
+ * \param file_path   The path to the file to read in.
+ * \return A new camera object representing the contents of the read-in file.
+ */
+camera_rpc_sptr
```

Comment:
I think the return type here should be a base class `camera_sptr`.  For now this just converts RPC metadata, but we should later port over the code from MAP-Tk for perspective cameras from metadata and put it in this same function.  The function would have to inspect the available metadata tags and determine which camera to create.  We don't need to do that on this branch, but using the base camera as a return type now will allow for this change later without needing to change the API.